src/game.rs

```diff
@@ -413,6 +439,46 @@ impl Game {
         }
     }
 
+    /// in order to update the door to be a clear wall, we need a few things:
+    /// 1. we need a reference to the tile in self.tiles that contains the heister_square
+    /// 2. we need to be able to know which wall on which square  to update
+    /// 3. we need to replace that square wth one who has a clear wall instead of a door
+    fn open_door(&mut self, door_pos: MapPosition, src_square: Square, dir: &MoveDirection) -> () {
+        let current_tile_position = Self::current_tile_position(&door_pos, &dir);
+        let mut tile = &mut Tile::default();
+        for t in &mut self.game_state.tiles {
+            if t.position == current_tile_position {
+                tile = t;
+                break;
+            }
+        }
+        if tile.squares.len() == 0 {
+            panic!("No tile found at pos {:?}", door_pos);
+        }
+
+        for mut square in &mut tile.squares {
+            if square == &src_square {
+                // Open The Door
+                match dir {
+                    MoveDirection::North => {
+                        square.north_wall = WallType::Clear;
+                    }
+                    MoveDirection::East => {
+                        square.east_wall = WallType::Clear;
+                    }
+                    MoveDirection::South => {
+                        square.south_wall = WallType::Clear;
+                    }
+                    MoveDirection::West => {
+                        square.west_wall = WallType::Clear;
+                    }
+                }
+                return;
+            }
+        }
+        panic!("When opening a door, we expect the square to have a door to open");
```

Comment:
This should probably return an error instead of panicking but I dont care so I will merge this.